crates/e2e/macro/src/config.rs

```diff
@@ -77,10 +76,12 @@ impl E2EConfig {
         self.backend.clone()
     }
 
-    /// The URL to the running node. If not set then a default node instance will be
-    /// spawned per test.
+    /// The URL to the running node, default to `CONTRACTS_NODE_URL` if not set.
+    /// If no URL is provided, then a default node instance will be spawned per test.
     pub fn node_url(&self) -> Option<String> {
-        self.node_url.clone()
+        self.node_url
+            .clone()
+            .or_else(|| std::env::var("CONTRACTS_NODE_URL").ok())
```

Comment:
Should it not be the other way around, the env var has precendence so you can override the hardcoded URL?